### ISSUE TITLE
Implement `@customName` extension

### ISSUE THREAD

**Issue by Cheatoid** [TypeScriptToLua/TypeScriptToLua#1498]
Occasionally there are times where you want to be able to declare a *global* function which also happens to be TS reserved keyword (e.g. `import`, `delete`). But, this means you are unable to call such Lua function, that is without utilizing tstl plugin or a bit of cleverness to workaround such restriction (which is just ugly), the following is an example workaround for invoking global `import` function:
[Playground](https://typescripttolua.github.io/play/#code/PTAEEEBNNAXALAlgZ1Mg9gWwKYHd7YBO2oiAdqAJ7oCuhoAKgMoMAyoADoegFbYDGsAFwAoAGY0ygxOgoASRJg7pCsADwMAfAAoEKIaABu6RJAA0oMgEMcB5LELkA5gEoDDUAG8RoX6GKwdBTa2gD6AOKgVqhWZJRRqKw0VgxWAEYANtguAHRO2LDaAESKyqpFLglwlBzY6GKgCkoqsC7a1jguANwiAL4iIiCgAKIAHjYcWaA0yFb5ouSwRGJW-CQM2PYAsuiQNFPefqDY40pZukjIBsamFlYGZDSYaURuaA7OPf38svZwm7BQABeRqlFoaAE7PZZHRFJb2Co9eGwHInCbnACMAAYMd0gA).

And at some other times, you just feel like renaming Lua call expression/identifier while preserving TS naming (perhaps for better readability, or to remain consistent with the rest of your TS codebase), take this illustrative example:
Input TS:
```typescript
declare namespace net {
    /** @customName get_version */
    export function GetVersion(this: void): number;
}
// Usage:
print(net.GetVersion());
```
Expected transpiled Lua:
```lua
print(net.get_version())
```

The `@customName` extension would allow us to get around the reserved keyword restriction mentioned earlier.
I propose this extension to be added as part of TSTL language extensions out-of-box, because I think it would be future-proof and more beneficial for the users (as opposed to me implementing this through [npm package](https://github.com/Cheatoid/TSTL-extensions)).

**Comment by Z3rio** [comment]
This sir, is fucking genius.

I created a custom plugin to solve this exact issue, but your solution for making it "built-in", would be even better.
I'll have a look at implementing this :)

**Comment by Z3rio** [comment]
@Perryvw you can go ahead and assign me to this if you wish, I've started having a look at it.

**Comment by Z3rio** [comment]
@Perryvw this can be closed as completed

**Comment by Perryvw** [comment]
Delivered with #1505, I'll make a new release today

**Comment by colinmcardell** [comment]
This is a really helpful extension. Thanks for implementing it.

I'm curious how to make it work for a property, getter, or setter on a class. Should this just work, or is it not supported as implemented in #1505? 

For example:

```typescript
class MyClass {
    /**
     * @customName my_foo
     */
    myFoo: number

    /**
     * @customName my_bar
     */
    get myBar(): number

    /**
     * @customName my_baz
     */
    set myBaz(v: number)
}
```

**Comment by Z3rio** [comment]
> This is a really helpful extension. Thanks for implementing it.
> 
> I'm curious how to make it work for a property, getter, or setter on a class. Should this just work, or is it not supported as implemented in #1505?
> 
> For example:
> 
> ```ts
> class MyClass {
>     /**
>      * @customName my_foo
>      */
>     myFoo: number
> 
>     /**
>      * @customName my_bar
>      */
>     get myBar(): number
> 
>     /**
>      * @customName my_baz
>      */
>     set myBaz(v: number)
> }
> ```

That should work, yeah, if it doesn't, then try to make it a one-lined comment just as in our example, instead of multiline.

**Comment by Perryvw** [comment]
I think `@customName` only works on separate declarations (ie of variables of functions). They do not seem to apply to object members at the moment.
